src/community/label_propagation.jl

```diff
@@ -0,0 +1,106 @@
+"""
+Community detection using the label propagation algorithm (see [Raghavan et al.](http://arxiv.org/abs/0709.2938)).
+`g`: imput Graph
+`maxiter`: maximum number of iterations
+return : vertex assignments and the convergence history
+"""
+function label_propagation(g::SimpleGraph; maxiter=1000)
+    n = nv(g)
+    label = collect(1:n)
+    runing_nodes = Set(vertices(g))
+    c = NeighComm(collect(1:n), fill(-1,n), 1)
+    convergence_hist = Vector{Int}()
+    random_order = Array(Int, n)
+    i = 0
+    while !isempty(runing_nodes) && i < maxiter
+        num_active = length(runing_nodes)
+        push!(convergence_hist, num_active)
+        i += 1
+        
+        # processing nodes in random order
+        for (j,node) in enumerate(runing_nodes)
+            random_order[j] = node
+        end
+        range_shuffle!(1:num_active, random_order)
+        @inbounds for j=1:num_active
+            u = random_order[j]
+            old_comm = label[u]
+            label[u] = vote!(g, label, c, u)
+            if old_comm != label[u]
+                for v in out_neighbors(g, u)
+                    push!(runing_nodes, v)
+                end
+            else
+                delete!(runing_nodes, u)
+            end
+        end
+    end
+    fill!(c.neigh_cnt, 0)
+    renumber_labels!(label, c.neigh_cnt)
+    label, convergence_hist
+end
+
+type NeighComm
+  neigh_pos::Vector{Int}
+  neigh_cnt::Vector{Int}
+  neigh_last::Int
+end
+
+function range_shuffle!(r::UnitRange, a::AbstractVector)
+    (r.start > 0 && r.stop <= length(a)) || error("out of bounds")
+    @inbounds for i=length(r):-1:2
+        j = rand(1:i)
```

Comment:
@afternone you can see some example of the seed optional argument in `randgraphs.jl`. It is better to have a seed optional argument here to be consistent to be consistent with the API, and because sometimes one wants to use one function with a fixed seed and the next one with a random seed, so seeding the global random generator and using only that could be a solution not flexible enough.